reflect CLI options on web URL

thisiscam opened the pull request:
Attempt at #50 .

Currently serializes CLI options with JSON, which is totally unreadable.

FlamingTempura commented on docs/index.ts:
```diff
@@ -198,6 +199,60 @@ function getOptions(): Options {
+function getOptionsFromURL() : Options {
+	const queryString = window.location.search;
+	const urlParams = new URLSearchParams(queryString);
+	const options_json = urlParams.get('cli');
```
```suggestion
	const options_json = urlParams.get('opt');
```

FlamingTempura commented on docs/index.ts:
```diff
@@ -198,6 +199,60 @@ function getOptions(): Options {
+function updateURLParams() {
+	const options = getOptions();
+	const options_json = JSON.stringify(options);
+	window.history.pushState(options, "", `index.html?cli=${encodeURIComponent(options_json)}`);
```
```suggestion
	window.history.pushState(options, "", `index.html?opt=${encodeURIComponent(options_json)}`);
```

thisiscam replied:
Fixed in the new commit. Thanks!

FlamingTempura reviewed (changes requested):
Thanks for the PR!

The only thing I'd like changed is to call the query param `opt` instead of `cli` (since it's not really options specific to the command line).

Other than that, good to go! Really appreciate you working on this feature (I don't find much time to maintain this library).

I'll make the merge when this change is made.

thisiscam commented:
@FlamingTempura I implemented your suggestion and added a delay to textarea inputs, to avoid accumulating a browser history time frame for each character one type.

FlamingTempura commented:
Merged, thanks for your contribution
